hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsVolumeList.java

```diff
@@ -291,6 +304,11 @@ public String toString() {
   void addVolume(FsVolumeReference ref) {
     FsVolumeImpl volume = (FsVolumeImpl) ref.getVolume();
     volumes.add(volume);
+    if (enableSameDiskTiering &&
```

Comment:
maybe consider putting this check into a method

Reply:
Good idea, will do that